docs/developer-guide/parallel-execution.md

```diff
@@ -0,0 +1,19 @@
+---
+description: StreamingFast Substreams parallel execution
+---
+
+# Parallel execution
+
+Parallel execution is the process of a Substreams module's code executing multiple segments of blockchain data simultaneously. Substreams modules are executed in parallel, rapidly producing data for consumption in end-user applications. Parallel Execution enables Substreams' highly effecient blockchain data processing capabilities.
+
+Parallel execution occurs when a requested module's start block is further back in the blockchain's history than the requested start block. For example, if a module starts at block 12,000,000 and a user requests data at block 15,000,000, parallel execution is used. This applies to both the development and production modes of Substreams operation. Parallel execution is performed for the full range of blocks when Substreams is in production mode.
```

Comment:
It can happen in two ways, the one described here but also in forward parallel execution, the first sentence gives the impression that only the former is true.

I see the last sentence of this paragraph talking about full range, but it's really not clear. I think a re-work is neeed.

Reply:
First sentence updated. I removed the last sentence. I do not know how to improve it.

"Parallel execution is performed for the full range of blocks when Substreams is in production mode."

Please let me know if you have any input to reword this for clarity and technical accuracy.

Thank you!

The changes will be available for review in the next commit/push.